src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesObject = {};
+  const stylesArray = sourceString.split(';');
+
+  stylesArray.forEach((styles) => {
+    let style = styles.split(':');
+
+    style = style.map(e =>
+      e.toString(e)
+        .trim()
+        .replace(/[\n]/g, ''))
+      .filter(e => e.length !== 0);
+
+    if (style[0] !== undefined) {
+      stylesObject[style[0]] = style[1];
+    }
+  });
+
+  return stylesObject;
```

Comment:
You can simplify your code a bit

```suggestion
  const stylesObject = {};
  const stylesArray = sourceString
    .split(';')
    .filter(e => e.trim());

  stylesArray.forEach((styles) => {
    let style = styles.split(':');

    style = style.map(e => e.trim());
    
     stylesObject[style[0]] = style[1];
    }
  });

  return stylesObject;
```